sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/PodInstance.java

```diff
@@ -22,6 +22,10 @@ default String getName() {
     }
 
     static String getName(PodSpec podSpec, int index) {
-        return String.format("%s-%d", podSpec.getType(), index);
+        return getName(podSpec.getType(), index);
+    }
+
+    public static String getName(String podType, int index) {
```

Comment:
It's called at PodsResource.java:L306, where only the launched TaskInfo is available.